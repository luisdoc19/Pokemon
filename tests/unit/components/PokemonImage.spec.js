import PokemonImageVue from "@/components/PokemonImage.vue";
import { shallowMount } from "@vue/test-utils";

describe("PokemonImage", () => {
  it("debe hacer match con el snapShot", () => {
    const wrapper = shallowMount(PokemonImageVue, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("debe mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = shallowMount(PokemonImageVue, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });
    const [img1, img2] = wrapper.findAll("img");

    expect(img1.exists()).toBeTruthy();
    expect(img2).toBe(undefined);
    expect(img1.classes("pokemon-hidden")).toBeTruthy();
  });

  it("debe mostrar la imagen true y el pokemon 100", () => {
    const pokemonId = 100;
    const wrapper = shallowMount(PokemonImageVue, {
      props: {
        pokemonId,
        showPokemon: true,
      },
    });
    const img1 = wrapper.find("img");

    expect(img1.attributes().src).toBe(
      `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonId}.svg`
    );
    expect(img1.exists()).toBeTruthy();
  });
});
