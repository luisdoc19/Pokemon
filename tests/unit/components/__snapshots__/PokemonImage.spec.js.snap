// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage debe hacer match con el snapShot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="pokemon-hidden" alt="pokemon"></div>`;
